python/cog/schema.py

```diff
@@ -43,7 +44,14 @@ class PredictionBaseModel(pydantic.BaseModel, extra=pydantic.Extra.allow):
 
 
 class PredictionRequest(PredictionBaseModel):
-    id: t.Optional[str]
+    # there's a problem here where the idempotent endpoint is supposed to
+    # let you pass id in the route and omit it from the input
+    # however this fills in the default
+    # maybe it should be allowed to be optional without the factory initially
+    # and be filled in later
+    #
+    # actually, this changes the public api so we should really do this differently
```

Comment:
If this does in fact change the public API of Cog, then this comment needs addressing before this can merge.

Reply:
turns out this isn't needed yet, it only becomes necessary once we start tracking multiple predictions